test/conftest.py

```diff
@@ -161,6 +161,12 @@ def reset(self):
     yield controller
     controller.reset()
 
+    # delete proxy
+    # allow some grace period to ensure we don't remove the proxy before
+    # other fixtures have torn down
+    resource = 'http://{}/proxies/{}'.format(toxiproxy_server, proxy_name)
+    eventlet.spawn_after(10, requests.delete, resource)
```

Comment:
does this mean that last one of these might keep the test run from finishing for 10 seconds?

Reply:
No, a process won't wait for greenthreads to finish before exiting.

This cleanup really isn't required because we stop the whole toxiproxy server at the end of the test run. I added it while debugging a hanging toxiproxy (which turned out to be http://tanin.nanakorn.com/blogs/373) thinking that the problem _might_ be the growing number of proxies that we add and never remove.

Reply:
I guess not. Will remove